src/opers.c

```diff
@@ -944,6 +957,204 @@ Obj FuncWITH_HIDDEN_IMPS_FLAGS(Obj self, Obj flags)
     return with;
 }
 
+
+/****************************************************************************
+**
+*F  FuncRankFilter( filt ) . . . . . . .  return rank of filter or flags
+*/
+Obj RANK_FILTERS;
+Obj FuncRankFilter(Obj self, Obj flags)
+{
+    Int res, i, lenrf, pos;
+    Obj impflags, trues;
+
+    /* argument check and maybe first get flags */
+    if ( IS_OPERATION(flags) ) {
+        flags = FuncFLAGS_FILTER(0, flags);
+    }
+    while ( TNUM_OBJ(flags) != T_FLAGS ) {
+            flags = ErrorReturnObj( "<flags> must be a flags list (not a %s)",
+            (Int)TNAM_OBJ(flags), 0L,
+            "you can replace <flags> via 'return <flags>;'" );
+    }
+    impflags = FuncWITH_HIDDEN_IMPS_FLAGS(0, flags);
+    trues = FuncTRUES_FLAGS(0, impflags);
+    if (trues == 0) return INTOBJ_INT(0);
+    lenrf = LEN_PLIST(RANK_FILTERS);
+    for (res = 0, i = 1; i <= LEN_PLIST(trues); i++) {
+        pos = INT_INTOBJ(ELM_PLIST(trues, i));
+        if (pos > lenrf || ELM_PLIST(RANK_FILTERS, pos) == 0)
+            res++;
+        else
+            res += INT_INTOBJ(ELM_PLIST(RANK_FILTERS, pos));
+    }
+    return INTOBJ_INT(res);
+}
+
+static Obj IMPLICATIONS_SIMPLE;
+static Obj IMPLICATIONS_COMPOSED;
+static Obj WITH_IMPS_FLAGS_CACHE;
+static const Int imps_cache_length = 11001;
```

Comment:
More idiomatic would be to say
```C
enum {
  imps_cache_length = 11001
};
```
Although modern C compilers should optimize both to the same output *shrug*

More concerning is the use of an all-lowercase name for a global, which I really don't like -- in virtually all of the rest of the GAP kernel, an all-lowercase name denotes a local variable.

So perhaps use `IMPS_CACHE_LENGTH` instead.